### ISSUE TITLE
about some training details~

### ISSUE THREAD

**Issue by Feynman1999** [spacepxl/WanTraining#3]
I noticed your repo:  https://huggingface.co/spacepxl/Wan2.1-control-loras. 
I have some questions about the training details (deblur):
1. What are the base learning rate and input_lr_scale?
2. How are prompts extracted? Is the qwenVL model used?

**Comment by spacepxl** [comment]
1. I changed LR several times depending on the specific run, but 1e-5 or 5e-6 seemed to be good for a rank 128 lora. 

    Input lr scale was 10 initially for the tile control lora, but that was probably too high, and I lowered it to 5, then eventually 1 for depth. Higher scales were helpful for sd1.5 but maybe not necessary for Wan.

2. Initially I used florence2 for captions and that was fine for tile, but with the depth lora I noticed that prompt following wasn't great. I've recaptioned most of my dataset with qwen now but haven't retrained with it yet.

3. Shift at inference vs training don't need to use the same values. I trained the tile lora with shift=1 because blur is a strong control signal and low shift gives more training priority to details. For depth, I had to raise shift to 5 to get it to learn to use the depth signal instead of ignoring it. Higher shift means more effort on learning structure and less on details.

**Comment by Feynman1999** [comment]
> 1. I changed LR several times depending on the specific run, but 1e-5 or 5e-6 seemed to be good for a rank 128 lora.
>    Input lr scale was 10 initially for the tile control lora, but that was probably too high, and I lowered it to 5, then eventually 1 for depth. Higher scales were helpful for sd1.5 but maybe not necessary for Wan.
> 2. Initially I used florence2 for captions and that was fine for tile, but with the depth lora I noticed that prompt following wasn't great. I've recaptioned most of my dataset with qwen now but haven't retrained with it yet.
> 3. Shift at inference vs training don't need to use the same values. I trained the tile lora with shift=1 because blur is a strong control signal and low shift gives more training priority to details. For depth, I had to raise shift to 5 to get it to learn to use the depth signal instead of ignoring it. Higher shift means more effort on learning structure and less on details.
Thank you for taking the time to answer amidst your busy schedule. Your response is very informative. I also want to ask, what is the value of controllinject_noise; Also, did you mention a learning rate of BATCHsize=1 for training?

**Comment by Feynman1999** [comment]
> 1. I changed LR several times depending on the specific run, but 1e-5 or 5e-6 seemed to be good for a rank 128 lora.
>    Input lr scale was 10 initially for the tile control lora, but that was probably too high, and I lowered it to 5, then eventually 1 for depth. Higher scales were helpful for sd1.5 but maybe not necessary for Wan.
> 2. Initially I used florence2 for captions and that was fine for tile, but with the depth lora I noticed that prompt following wasn't great. I've recaptioned most of my dataset with qwen now but haven't retrained with it yet.
> 3. Shift at inference vs training don't need to use the same values. I trained the tile lora with shift=1 because blur is a strong control signal and low shift gives more training priority to details. For depth, I had to raise shift to 5 to get it to learn to use the depth signal instead of ignoring it. Higher shift means more effort on learning structure and less on details.

Thank you for taking the time to answer amidst your busy schedule. Your response is very informative. I also want to ask, what is the value of controllinject_noise; Also, did you mention a learning rate of BATCHsize=1 for training?

**Comment by spacepxl** [comment]
Correct, those are for batch size = 1, currently the training script does not support larger batch sizes because it would require accounting for mixed resolutions.

Noise injection was initially 0, but I increased it to 0.1 for the last training run on tile to allow more creativity.

**Comment by Feynman1999** [comment]
> Correct, those are for batch size = 1, currently the training script does not support larger batch sizes because it would require accounting for mixed resolutions.
> 
> Noise injection was initially 0, but I increased it to 0.1 for the last training run on tile to allow more creativity.

ok！ Does cfg have a significant impact on inference? I tried to replicate the training settings of tile, and when cfg is large, there will be more artifacts (unreasonable generation)

**Comment by spacepxl** [comment]
I've seen good results with CFG in the 4-6 range same as the base model, but sampler could also play a role.
